src/ejabberd_cache_mnesia.erl

```diff
@@ -0,0 +1,175 @@
+%%%-------------------------------------------------------------------
+%%% File    : ejabberd_cache_mnesia.erl
+%%% Author  : Devin Sills <dsills@skillz.com>
+%%% Created : 18 Jan 2022 by Devin Sills <dsills@skillz.com>
+%%%
+%%%----------------------------------------------------------------------
+
+-module(ejabberd_cache_mnesia).
+
+-behaviour(gen_server).
+
+%% External exports
+-export([start_link/2, get_proc_name/1]).
+
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2, terminate/3, handle_info/3]).
+
+-include("logger.hrl").
+
+start_link(CacheName, MaxCacheSize) ->
+  %% sql cache table
+  mnesia:create_table(get_cache_table_name(CacheName), [{attributes, [ key, value ]}, {type, set}, {ram_copies, [node()]}, {local_content, false}]),
```

Comment:
i'm not sure this makes sense to run each time this process is created, is there another spot where mnesia schemas are setup?

Reply:
the table name is dependent on the cache name, which is sent in as a parameter to the start_link function, which is why I placed it here. If we consolidated the cache tables into one spot outside of this module, then we'd have to pre-create all of the cache tables whereas IMO it's more flexible to just create the table with the process so that caches can be created anywhere by creating the process via start_link. If the table already exists, it does error out but just returns w/ a response that the table is already created.